src/compiler/checker.ts

```diff
@@ -35198,6 +35180,12 @@ namespace ts {
             }
         }
 
+        function getPropertyNameForKnownSymbolName(symbolName: string): __String {
+            const ctorType = getGlobalESSymbolConstructorSymbol(/*reportErrors*/ false);
+            const uniqueType = ctorType && getTypeOfPropertyOfType(getTypeOfSymbol(ctorType), escapeLeadingUnderscores(symbolName));
+            return uniqueType && isTypeUsableAsPropertyName(uniqueType) ? getPropertyNameFromType(uniqueType) : `__@${symbolName}` as __String;

```

Comment:
The `__@${symbolName} as __String;` case is reachable in scenarios where our internals request a name for a symbol which doesn't exist in the lib. Eg, requesting `Symbol.asyncIterator` in a `lib` target where it doesn't exist. No construct can _make_ a property symbol name in that format anymore (unique symbol names have the form `__@desc@###`), so it's guaranteed to result in an `undefined` result from `getPropertyOfType`, resulting in a `noIterationTypes` result (as you may expect when the core iteration protocol types are missing!).

Reply:
Can you give a little more context on the significance? Are you saying this is handled well in the existing code, that other code needs to be careful, that there's a fundamental architectural issue, or that this now has a user-facing impact? (or all?)

Reply:
I'm saying that properties with a name like that can no longer be constructed, so when the proper name cannot be found, returning a name like that will always make a follow-up property lookup fail, which has the desired outcome.